src/electron/renderer/dom.ts

```diff
@@ -97,9 +97,14 @@ function appendTestResultDOM(r) {
 
   if (!title) return;
 
+  let code = r.failureMessage ? r.failureMessage : '';
+
   const ts = r.testResults.map((tr) => {
     const { title, status, duration, failureMessages } = tr;
-    const code = Array.isArray(failureMessages) ? failureMessages[0] : '';
+
+    if (!code) {
+      code = Array.isArray(failureMessages) ? failureMessages[0] : '';
+    }
```

Comment:
whether the code is on r. failureMessage, or on tr. failureMessages?

Reply:
source code lines exist within `r.failureMessage`,  `tr. failureMessages[0]` is assigned to variable `code` if `r.failureMessage` is empty/undefined